### PR TITLE
Add detection for CVE-2024-50379

### PR DESCRIPTION
**Pull request by nmasdoufi-ol**
image

### CODE REVIEW

File: agent/exploits/cve_2024_50379.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    accept_pattern = [re.compile(r"Apache Tomcat")]
+
+    def check(self, target: definitions.Target) -> list[definitions.Vulnerability]:
+        vulnerabilities: list[definitions.Vulnerability] = []
+
+        try:
+            root_response = requests.get(
+                target.origin, timeout=DEFAULT_TIMEOUT.seconds, verify=False
+            )
+            if root_response.status_code != 200:
```
Comment (amine3) [inline agent/exploits/cve_2024_50379.py:65]: Why only 200?

File: agent/exploits/cve_2024_50379.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                data="<!-- test -->",
+                timeout=DEFAULT_TIMEOUT.seconds,
+                verify=False,
+            )
+
+            if (
+                test_response.status_code in (201, 204)
+                or _is_version_vulnerable(tomcat_version) is True
+            ):
+                vulnerability = self._create_vulnerability(target)
```
Comment (amine3) [inline agent/exploits/cve_2024_50379.py:89]: vulnerabilities = [self._create_vulnerability(target)] since you have one iterration

**Review by amine3 — CHANGES_REQUESTED** [review]

File: agent/exploits/cve_2024_50379.py (hunk 2 above)
Comment (nmasdoufi-ol) [reply agent/exploits/cve_2024_50379.py:89]: Sure.

File: agent/exploits/cve_2024_50379.py (hunk 1 above)
Comment (nmasdoufi-ol) [reply agent/exploits/cve_2024_50379.py]: You're right, it's too restrictive.

**Review by amine3 — APPROVED** [review]

**Pull request merged**